src/iotivity/iotivity_client.cc

```diff
@@ -0,0 +1,463 @@
+/****************************************************************************
+**
+** Copyright © 1992-2014 Cisco and/or its affiliates. All rights reserved.
+** All rights reserved.
+**
+** $CISCO_BEGIN_LICENSE:APACHE$
+**
+** Licensed under the Apache License, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+** http://www.apache.org/licenses/LICENSE-2.0
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+** $CISCO_END_LICENSE$
+**
+****************************************************************************/
+#include <string>
+#include <map>
+
+#include "iotivity/iotivity_client.h"
+#include "iotivity/iotivity_device.h"
+#include "iotivity/iotivity_resource.h"
+
+
+IotivityClient::IotivityClient(IotivityDevice* device) : m_device(device) {
+    m_asyncCallId_findresources = 0;
+    m_asyncCallId_finddevices = 0;
+    m_discoveryOptionDeviceId = "";
+}
+
+IotivityClient::~IotivityClient() {
+    for (auto const &entity : m_resourcemap) {
+        IotivityResourceClient *resClient =
+            reinterpret_cast<IotivityResourceClient *>(entity.second);
+        delete resClient;
+    }
+    m_resourcemap.clear();
+    m_foundresourcemap.clear();
+}
+
+void *IotivityClient::getResourceById(std::string id) {
+    if (m_resourcemap.size()) {
+        std::map<std::string, void *>::const_iterator it;
+        if ((it = m_resourcemap.find(id)) != m_resourcemap.end())
+            return reinterpret_cast<void *>((*it).second);
+    }
+
+    return NULL;
+}
+
+void IotivityClient::foundResourceCallback(std::shared_ptr<OCResource> resource,
+        int waitsec) {
+    DEBUG_MSG("foundResourceCallback:\n");
+    bool mapFoundResource = true;
+
+    IotivityResourceClient *resClient = new IotivityResourceClient(m_device);
+    resClient->setSharedPtr(resource);
+    PrintfOcResource((const OCResource &)*resource);
+
+    if (m_discoveryOptionDeviceId != "") {
+        if (resource->sid() != m_discoveryOptionDeviceId) {
+            mapFoundResource = false;
+        }
+    }
+
+    if (mapFoundResource) {
```

Comment:
Change to `if (m_discoveryOptionDeviceId == "" || resource->sid() == m_discoveryOptionDeviceId)`, then you can get rid of `mapFoundResource` and line 64~68